### ISSUE TITLE
Exponential backoff for vibrations

### ISSUE THREAD

**Issue by ruuda** [signalapp/Signal-Android#5945]
I have:
- [x] Searched open and closed issues for duplicates. #3165 and #5813 are related.
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
When a contact is rambling at you for say ~20 messages in short succession, the phone will vibrate every time. This is annoying. It looks like 3787551878473c703a21210aa24d39080ad57ad4 introduced a limit of once ever two seconds, but that will not help a lot when a contact keeps sending messages for a few minutes.

When my phone vibrates, I have to make a decision:
- Stop what I was doing and look at the message right now.
- Continue what I was doing, remember that somebody sent me a message, and take a look when I have the time.

Most of the time, I’ll opt for the second option. Then again, there are two cases:
- After a few minutes I remember to take a look.
- I forget that I received a notification and I don’t look at my phone until it starts vibrating again.

If my phone vibrates twice within a few seconds, that is not useful, it is only annoying. But even with a two-second delay, if I decide to not look *right now*, a vibration after 2, 3, 6, ... seconds is still annoying. I already made a mental note to take a look soon, please stop bothering me. On the other hand, if after say ~15 minutes I receive another message, then that is probably not part of the same discussion, and if I haven’t checked by then, I do want a notification.

I think that an exponential delay might work well here. When a message is received, vibrate only if the last vibration was more than *x* seconds ago. Start with *x* = 2 seconds, and double (or maybe even triple) it at every vibration, up to a maximum of e.g. 5 minutes. When I read a message, reset *x* to its initial value.

### Steps to reproduce
- Have a contact send you a message every few seconds.
- Put the phone on the table, screen locked.

**Actual result:** The phone keeps on vibrating while the contact is sending its monologue. Currently (without the 2-second delay) this means the phone can be vibrating continuously for minutes.
**Expected result:** The phone should vibrate once, maybe twice, and then remain silent for a while.

### Device info
**Device:** Nexus 6P
**Android version:** 7.0, 5 November 2016 patch level
**Signal version:** 3.24.1

**Comment by jeremymasters** [comment]
One thing that could reduce it would be to turn off in-conversation notifications in the settings.

**Comment by ruuda** [comment]
Thank you for the suggestion. I had already turned that setting off, but in the scenario where this is an issue, I am not in a conversation; this is about notifications/vibrations when the phone is locked.

**Comment by wchristian** [comment]
Something i'd expect to see here:

1. Have notifications occur per conversation activation.
2. Have settings per conversation for how often it may notify at all. (every 10 minutes, hour, 6 hours, day, never, etc.)

And optimally, dunno if possible:
3. Don't send notifications for messages seen on other clients. (I'll have a convo on my pc and the phone rattles itself to death.)

**Comment by GSPP** [comment]
I think the delay strategy needs to be configurable. With most friends I'd be OK with receiving messages only after quite some time. When my boss writes me I want my phone to buzz immediately and possibly for every message also.

**Comment by deutrino** [comment]
Although it doesn't feel ideal, I would be happy with a per-conversation setting, and I can't think of a better way to do it. The setting should apply whether or not the phone is idle - I have chatty contacts, and the vibration gets really annoying when I'm using my phone for something else while also talking on Signal.

**Comment by automated-signal** [comment]
GitHub Issue Cleanup:
See #7598 for more information.
